### PR TITLE
Update PyROS Subproblem Initialization Routines

### PR DESCRIPTION
**Pull request by shermanjasonaf**
## Summary/Motivation

Implement updates to the PyROS subproblem initialization routines. These updates have been observed to improve the computational performance and reliability of PyROS for both small-scale and large-scale instances.

## Changes proposed in this PR:
- Initialize second-stage variables $z^k$ and state variables $y^k$ of master problem of iteration $k$ to their values in the worst-case separation solution of iteration $k - 1$. Consequently, only the performance constraints of master block $k$ are violated by the initial master point. Drop equality constraint slack variables from the slack-based master feasibility problem.
- Initialize separation problem for each performance constraint to variable values from master block with worst violation of the performance constraint. Hence, each separation problem is initialized to the best known solution (and provided the set is not of type `DiscreteScenarioSet`, the solution is feasible).
- Refactor decision rule variable and constraint declaration routines.
- Refactor decision rule polishing problem routine. Ensure auxiliary variables are initialized to the initial values of the quantities they are meant to represent, rather than the currently used blanket value of `1E6`.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by shermanjasonaf** [comment]
Feel free to add @natalieisenberg as a reviewer.

**Review by natalieisenberg — APPROVED** [review]
Look good! I like the new system for managing decision rules.

**Review by mrmundt — APPROVED** [review]
Same general comment as before that, if PyROS is ever promoted out of `contrib`, I will request that instances of `master` are changed to some other more inclusive language choice.

**Review by jsiirola — APPROVED** [review]
Two minor things, but nothing to prevent merging

File: pyomo/contrib/pyros/master_problem_methods.py [hunk 1]
```diff
@@ -123,62 +128,37 @@ def construct_master_feasibility_problem(model_data, config):
...
-            dr_eqs = list()
-
         targets.extend(
             [
                 con
                 for con in blk.component_data_objects(
                     Constraint, active=True, descend_into=True
                 )
-                if con not in dr_eqs
+                if not con.equality
```
[the remark this answers is not in the document]
Comment (shermanjasonaf) [reply pyomo/contrib/pyros/master_problem_methods.py:143]: As of #2206, ranged inequality constraints with identical bounds are recast to equality constraints during the preprocessing step by `util.transform_to_standard_form`.

**Pull request merged**